advisory_parser/parsers/mysql.py

```diff
@@ -66,37 +66,40 @@ def create_mariadb_cve_map():
 
 
 def parse_mysql_advisory(url):
+    # The url passed in can be either the main CPU page, or the "Text Form of
+    # Risk Matrices" (aka verbose) page
+
+    # Parse url first to get base url and cpu date
+    url_match = re.search(r'/cpu([a-z]{3})(\d{4})(verbose)?\.html(#.*)?$', url)
+    if not url_match:
+        raise AdvisoryParserTextException('Unexpected CPU URL format.')
+
+    # Get base url and determine advisory_url and verbose_url
+    url = url[0 : url_match.start() + len('/cpuMMMYYYY')]
```

Comment:
You can make the third and fourth capture group non-capturing and this can then use the same logic as before:

```python
In [5]: url_match = re.search(r'/cpu([a-z]{3})(\d{4})(?:verbose)?\.html(?:#.*)?$', url)

In [6]: day, month = url_match.groups()

In [7]: day, month
Out[7]: ('apr', '2019')
```

Note the addition of `?:` in the parentheses.

Reply:
Sure, good idea.  Corrected now.